k8s/operators/src/mayastorpool/client.rs

```diff
@@ -0,0 +1,86 @@
+use crate::{error::Error, mayastorpool::crd::MayastorPool};
+use kube::{
+    api::{DeleteParams, ListParams, Patch, PatchParams},
+    Api, Client, CustomResourceExt,
+};
+
+/// Get the MayastorPool api.
+pub fn api(client: &Client, namespace: &str) -> Api<MayastorPool> {
+    Api::namespaced(client.clone(), namespace)
+}
+
+/// Fetch list of all mayastor pools.
+pub async fn list(
+    client: &Client,
+    namespace: &str,
+    pagination_limit: u32,
+) -> Result<Vec<MayastorPool>, Error> {
+    // Create the list params with pagination limit.
+    let mut list_params = ListParams::default().limit(pagination_limit);
+    let pools_api: Api<MayastorPool> = api(client, namespace);
+    let mut pools: Vec<MayastorPool> = vec![];
+    loop {
+        let mut result = pools_api.list(&list_params).await?;
+        pools.append(&mut result.items);
+        // Check for the token, if valid then continue.
+        match result.metadata.continue_ {
+            Some(token) if !token.is_empty() => {
+                list_params = list_params.continue_token(token.as_str())
+            }
+            _ => break,
+        };
+    }
+
+    Ok(pools)
+}
+
+/// Delete all the mayastor pool CRs forcefully by patching finalizers.
+pub async fn delete(
+    client: &Client,
+    namespace: &str,
+    msps: &mut [MayastorPool],
+) -> Result<(), Error> {
+    // MayastorPool API.
+    let pools_api: Api<MayastorPool> = api(client, namespace);
+
+    // Create the api params with defaults.
+    let patch_params = PatchParams::default();
+    let delete_params = DeleteParams::default();
+    let list_params = ListParams::default();
+
+    // Patch all msps to remove the finalizers.
+    for msp in msps.iter_mut() {
+        msp.metadata.finalizers = Some(vec![]);
+        let name = msp.metadata.name.as_ref().ok_or(Error::InvalidCRField {
+            field: "crd.metadata.name".to_string(),
+        })?;
+        pools_api
+            .patch(name, &patch_params, &Patch::Merge(&msp))
+            .await?;
+    }
+
+    // Delete the whole msp cr collection from the cluster.
+    pools_api
+        .delete_collection(&delete_params, &list_params)
+        .await?;
+
+    Ok(())
+}
+
+/// Check whether the MayastorPool crd exists or not.
+pub async fn check_crd(k8s: &Client) -> Result<bool, Error> {
+    let crd_api: Api<k8s_openapi::apiextensions_apiserver::pkg::apis::apiextensions::v1::CustomResourceDefinition> = Api::all(k8s.clone());
```

Comment:
wow that's long, should we shorten it with a use? :D

Reply:
Yeah, we can :-p

Reply:
I did it in the other module but forgot here. 😅